improver_tests/lightning/test_LightningMultivariateProbability.py

```diff
@@ -0,0 +1,274 @@
+        units="J kg-1",
+        time=datetime(2017, 11, 10, 3, 0),
+        time_bounds=None,
+        attributes=None,
+        standard_grid_metadata="gl_ens",
+        domain_corner=(-60, 0),
+        grid_spacing=20,
+    )
+    return cube
+
+
+@pytest.fixture(name="liftidx_cube")
+def liftidx_cube_fixture() -> Cube:
+    """
+    Set up a liftidx cube for use in tests.
+    Has 2 realizations, 7 latitudes spanning from 60S to 60N and 3 longitudes.
+    """
+
+    data = np.full((2, 7, 3), dtype=np.float32, fill_value=10)
+    cube = set_up_variable_cube(
+        data,
+        name="temperature_difference_between_ambient_air_and_air_lifted_adiabatically",
+        units="K",
+        time=datetime(2017, 11, 10, 3, 0),
+        time_bounds=None,
+        attributes=None,
+        standard_grid_metadata="gl_ens",
+        domain_corner=(-60, 0),
+        grid_spacing=20,
+    )
+    return cube
+
+
+@pytest.fixture(name="pwat_cube")
+def pwat_cube_fixture() -> Cube:
+    """
+    Set up a pwat cube for use in tests.
+    Has 2 realizations, 7 latitudes spanning from 60S to 60N and 3 longitudes.
+    """
+
+    data = np.full((2, 7, 3), dtype=np.float32, fill_value=3)
+    cube = set_up_variable_cube(
+        data,
+        name="precipitable_water",
+        units="kg m-2",
+        time=datetime(2017, 11, 10, 3, 0),
+        time_bounds=None,
+        attributes=None,
+        standard_grid_metadata="gl_ens",
+        domain_corner=(-60, 0),
+        grid_spacing=20,
+    )
+    return cube
+
+
+@pytest.fixture(name="apcp_cube")
+def apcp_cube_fixture() -> Cube:
+    """
+    Set up a apcp cube for use in tests.
+    Has 2 realizations, 7 latitudes spanning from 60S to 60N and 3 longitudes.
+    """
+
+    data = np.full((2, 7, 3), dtype=np.float32, fill_value=6)
+    cube = set_up_variable_cube(
+        data,
+        name="precipitation_amount",
+        units="kg m-2",
+        time=datetime(2017, 11, 10, 6, 0),
+        time_bounds=(datetime(2017, 11, 10, 3, 0), datetime(2017, 11, 10, 6, 0)),
+        attributes=None,
+        standard_grid_metadata="gl_ens",
+        domain_corner=(-60, 0),
+        grid_spacing=20,
+    )
+    return cube
+
+
+@pytest.fixture(name="expected_cube")
+def expected_cube_fixture() -> Cube:
+    """
+    Set up the Lightning cube that we expect to get from the plugin
+    """
+
+    data = np.full((2, 7, 3), dtype=np.float32, fill_value=14.111012)
+    cube = set_up_variable_cube(
+        data,
+        name="20_km_lightning_probability_over_the_valid_time_of_the_accumulated_precipitation",
+        units="1",
+        time=datetime(2017, 11, 10, 4, 30),
+        time_bounds=(datetime(2017, 11, 10, 3, 0), datetime(2017, 11, 10, 6, 0)),
+        attributes=None,
+        standard_grid_metadata="gl_ens",
+        domain_corner=(-60, 0),
+        grid_spacing=20,
+    )
+    return cube
+
+
+def test_basic(cape_cube, liftidx_cube, pwat_cube, cin_cube, apcp_cube, expected_cube):
```

Comment:
Good catch. In my reworking of the unit tests, I've reworked and manually put in a range of values for each of the parameters (CAPE, CIN, precipitation accumulation, lifted index, and precipitable water) in the numpy arrays so that all the different conditions in the regression equations are now covered to ensure the expected output is produced for the wide range of conditions.

Specifically, instead of setting each of the arrays to fill values for the entire cube I've set the values to vary within each cube fixture, such as for CAPE:
`    data = np.array([[4000, 0], [0, 4000]], dtype=np.float32)`